cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerDecomissioner.java

```diff
@@ -99,27 +100,28 @@ public class ClouderaManagerDecomissioner {
     @Inject
     private FlowMessageService flowMessageService;
 
-    public void verifyNodesAreRemovable(Stack stack, Collection<InstanceMetaData> removableInstances, ApiClient client) {
+    public void verifyNodesAreRemovable(StackDtoDelegate stack, Collection<InstanceMetadataView> removableInstances, ApiClient client) {
         try {
             HostTemplatesResourceApi hostTemplatesResourceApi = clouderaManagerApiFactory.getHostTemplatesResourceApi(client);
             ApiHostTemplateList hostTemplates = hostTemplatesResourceApi.readHostTemplates(stack.getName());
 
-            for (HostGroup hostGroup : stack.getCluster().getHostGroups()) {
-                Set<InstanceMetaData> removableHostsInHostGroup = removableInstances.stream()
-                        .filter(instanceMetaData -> instanceMetaData.getInstanceGroup().getGroupName().equals(hostGroup.getName()))
+            for (InstanceGroupDto instanceGroupDto : stack.getInstanceGroupDtos()) {
```

Comment:
nope, the hostGropu contains the InstanceGroup so some cases to avoid the unnecessary re-fetchong of the instance group, I have changed the HostGroup to InstanceGroup where I could